misinformation/multimodal_search.py

```diff
@@ -0,0 +1,299 @@
+from misinformation.utils import AnalysisMethod
+import torch
+import torch.nn.functional as Func
+import requests
+import lavis
+from PIL import Image
+from IPython.display import display
+from lavis.models import load_model_and_preprocess
+
+
+class MultimodalSearch(AnalysisMethod):
+    def __init__(self, subdict: dict) -> None:
+        super().__init__(subdict)
+        # self.subdict.update(self.set_keys())
+
+    multimodal_device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+    def load_feature_extractor_model(device, model_type):
+        if model_type == "blip2":
```

Comment:
instead of if ... elif ... you can use a dictionary that maps each keyword to a function. Then call that mapped function, ie
```
select_case = { "blip2": set_model_blip2}
```
with
```
def set_model_blip2(device):
   ...
```
or use a dictionary mapping for all the values that are then handed to `load_model_and_preprocess)